File: OneCodeTeam/How-to-use-Azure-ACS-Access-Control-Service-27411/[VB]-How to use Azure ACS (Access Control Service) protect ASP.NET Web API/VB/VBAzureACSWithWebAPI/WebAPI/Controllers/ProductsController.vb
'***************************** Module Header ******************************\
' Module Name:	ProductsController.vb
' Project:		VBAzureACSWithWebAPI
' Copyright (c) Microsoft Corporation.
' 
' Secure Web API is a hot topic in asp.net forum.
' This sample will show you how to use Azure ACS secure the web API.
' It will require you sign in with Live ID/Google/Yahoo/Live connect account 
' first before you use the web API.
'
' ProductsController
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'**************************************************************************/
Imports System.Web.Http
Imports System.Net

Public Class ProductsController
    Inherits ApiController
    Private products As Product() = New Product() {New Product() With {
        .Id = 1, _
        .Name = "Tomato Soup", _
        .Category = "Groceries", _
        .Price = 1 _
    }, New Product() With { _
        .Id = 2, _
        .Name = "Yo-yo", _
        .Category = "Toys", _
        .Price = 3.75D _
    }, New Product() With { _
        .Id = 3, _
        .Name = "Hammer", _
        .Category = "Hardware", _
        .Price = 16.99D _
    }}

    Public Function GetAllProducts() As IEnumerable(Of Product)
        Return products
    End Function

    Public Function GetProductById(id As Integer) As Product
        Dim product = products.FirstOrDefault(Function(p) p.Id = id)
        If product Is Nothing Then
            Throw New HttpResponseException(HttpStatusCode.NotFound)
        End If
        Return product
    End Function

    Public Function GetProductsByCategory(category As String) As IEnumerable(Of Product)
        Return products.Where(Function(p) String.Equals(p.Category, category, StringComparison.OrdinalIgnoreCase))
    End Function

End Class

File: OneCodeTeam/How-to-use-Azure-ACS-Access-Control-Service-27411/[C#]-How to use Azure ACS (Access Control Service) protect ASP.NET Web API/C#/CSAzureACSWithWebAPI/WebAPI/Controllers/ProductsController.cs
/***************************** Module Header ******************************\
* Module Name:	ProductsController.cs
* Project:		CSAzureACSWithWebAPI
* Copyright (c) Microsoft Corporation.
* 
* Secure Web API is a hot topic in asp.net forum.
* This sample will show you how to use Azure ACS secure the web API.
* It will require you sign in with Live ID/Google/Yahoo/Live connect account 
* first before you use the web API.
*
* ProductsController is an example WebAPI controller.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\**************************************************************************/


using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Web.Http;
using WebAPI.Models;


namespace WebAPI.Controllers
{
    public class ProductsController : ApiController
    {
        Product[] products = new Product[] 
        { 
            new Product { Id = 1, Name = "Tomato Soup", Category = "Groceries", Price = 1 }, 
            new Product { Id = 2, Name = "Yo-yo", Category = "Toys", Price = 3.75M }, 
            new Product { Id = 3, Name = "Hammer", Category = "Hardware", Price = 16.99M } 
        };

        public IEnumerable<Product> GetAllProducts()
        {
            return products;
        }

        public Product GetProductById(int id)
        {
            var product = products.FirstOrDefault((p) => p.Id == id);
            if (product == null)
            {
                throw new HttpResponseException(HttpStatusCode.NotFound);
            }
            return product;
        }

        public IEnumerable<Product> GetProductsByCategory(string category)
        {
            return products.Where(
                (p) => string.Equals(p.Category, category,
                    StringComparison.OrdinalIgnoreCase));
        }

    }
}
